[LT] Move MakeNode into ir_builder.h

desertfire opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #76738
* #76711
* #76542
* __->__ #76482
* #76488

Summary: Move MakeNode into ir_builder.h to avoid circular header
reference later when introducing a trie cache for IR node lookup.

desertfire committed:
[LT] Move MakeNode into a separate header file

Summary: Move MakeNode and NodeCast into a separate ir_construct_util.h to avoid header cross-reference after adding Trie cache.

[ghstack-poisoned]

wconstab commented:
@desertfire you should take a look at https://github.com/pytorch/pytorch/pull/75433 which is close to landing, i think you should plan to land after it and maybe that means you would update this PR to use the 'ir_builder.h/.cpp.' files?

desertfire commented:
> @desertfire you should take a look at #75433 which is close to landing, i think you should plan to land after it and maybe that means you would update this PR to use the 'ir_builder.h/.cpp.' files?

I see. #75433 can actually simplify my later PRs, so please go ahead to land that first.

desertfire committed:
Update on "[LT] Move MakeNode into ir_builder.h"


Summary: Move MakeNode into ir_builder.h to avoid circular header
reference later when introducing a trie cache for IR node lookup.

[ghstack-poisoned]

desertfire committed:
Update on "[LT] Move MakeNode into ir_builder.h"


Summary: Move MakeNode into ir_builder.h to avoid circular header
reference later when introducing a trie cache for IR node lookup.

[ghstack-poisoned]

desertfire committed:
Update on "[LT] Move MakeNode into ir_builder.h"


Summary: Move MakeNode into ir_builder.h to avoid circular header
reference later when introducing a trie cache for IR node lookup.

[ghstack-poisoned]

desertfire commented:
@pytorchbot merge this
